### ISSUE TITLE
after() sees wrong response-code for missing routes.

### ISSUE THREAD

**Issue by ph-hs** [perwendel/spark#712]
1. Try this:
```java
    public static void main(String[] args) {
        get("/hello", (req, res) -> "Hello World");
        after((request, response) -> {
            System.out.println("Status: " + response.status());
        });
    }
```
2. In a browser hit, say, "/hello2" 

3. Note that it is 200 in the console but 404 in the browser.

It could be that there's something that happens after after() that's not open to participation :-( or a bug.

**Comment by ph-hs** [comment]
This is because filters registered through after(x) are called **before** httpResponse.setStatus(HttpServletResponse.SC_NOT_FOUND) is called in MatcherFilter.java

**Comment by tipsy** [comment]
Was this fixed in #546 @ph-hs ?

**Comment by tipsy** [comment]
Yes it was. Using `afterAfter` will give you 404.
